vyxal/helpers.py

```diff
@@ -44,13 +44,13 @@ def case_of(value: str) -> int:
 
 @lazylist
 def chop(it: VyIterable, n: int) -> LazyList:
-    """Chop `it` into `n` equal chunks, plus an optional extra piece for leftovers."""
+    """Chop `it` into `n` chunks."""
     chunk_len = len(it) // n
-
+    left_over = len(it) % n
     for i in range(n):
-        yield it[i * chunk_len : (i + 1) * chunk_len]
-    if n * chunk_len < len(it):
-        yield it[n * chunk_len :]
+        length = chunk_len + (1 if left_over > i else 0)
+        yield it[:length]
+        it = it[length:]
```

Comment:
If you want to golf it a bit, something like [this](https://tio.run/##jY5BCsIwEEX3PcVfJhopaVWk4ElKEWlTO1imIcZFTx/HglBwoX8zMP/NY/wch4nLkw8pda5HO0xeUTRgXWWQtMOT75fRMc5QMqTU2IKxg9XIc/CC9VMAgRjhyjenPtfvzOTGDhRrwmalq6BIRCJZbZss84E4qpEeUS3P1I3BXmv9XViDwqCU2uAglP2LKn5TBkcBSwFTegE) would work too, I think:
```python
chunk_len = (len(it) + n - 1) // n
for i in range(n):
    yield it[i * chunk_len : (i + 1) * chunk_len]
```

Reply:
I don't think it would, as this builtin divides into variable length chunks.